marlowe-playground-client/src/MainFrame/Types.purs

```diff
@@ -188,111 +186,105 @@ _walletSlot :: SProxy "walletSlot"
-  , activeJSDemo :: String
-  , showBottomPanel :: Boolean
-  -- TODO: rename to haskellEditorState
-  , haskellState :: HE.State
-  -- TODO: rename to javascriptEditorState
-  , javascriptState :: JS.State
-  , marloweEditorState :: ME.State
-  , simulationState :: Simulation.State
-  , projects :: Projects.State
-  , newProject :: NewProject.State
-  , rename :: Rename.State
-  , saveAs :: SaveAs.State
-  , authStatus :: WebData AuthStatus
-  , gistId :: Maybe GistId
-  , createGistResult :: WebData Gist
-  , loadGistResult :: Either String (WebData Gist)
-  , projectName :: String
-  , showModal :: Maybe ModalView
-  -- The source of truth for the unsaved change lives inside each editor
-  -- The only reason we store a copy of the state here is because of the render function
-  -- in the Mainframe view, which uses this derived state to show an unsaved indicator.
-  -- Inside the Mainframe view we can inspect the state of the submodules (Haskell/JS/Marlowe)
-  -- as their state is part of the MainFrame state, but we cannot inspect the state of the
-  -- child components Blockly and ActusBlockly as we need a Query.
-  , hasUnsavedChanges :: Boolean
-  -- The initial language selected when you create/load a project indicates the workflow a user might take
-  -- A user can start with a haskell/javascript example that eventually gets compiled into
-  -- marlowe/blockly and run in the simulator, or can create a marlowe/blockly contract directly,
-  -- which can be used interchangeably. This is used all across the site to know what are the posible
-  -- transitions.
-  , workflow :: Maybe Lang
-  }
-
-derive instance newtypeState :: Newtype State _
+type State
+  = { view :: View
+    , jsCompilationResult :: CompilationState
+    , jsEditorKeybindings :: KeyBindings
+    , activeJSDemo :: String
+    , showBottomPanel :: Boolean
+    -- TODO: rename to haskellEditorState
+    , haskellState :: HE.State
+    -- TODO: rename to javascriptEditorState
+    , javascriptState :: JS.State
+    , marloweEditorState :: ME.State
+    , simulationState :: Simulation.State
+    , projects :: Projects.State
+    , newProject :: NewProject.State
+    , rename :: Rename.State
+    , saveAs :: SaveAs.State
+    , authStatus :: WebData AuthStatus
+    , gistId :: Maybe GistId
+    , createGistResult :: WebData Gist
+    , loadGistResult :: Either String (WebData Gist)
+    , projectName :: String
+    , showModal :: Maybe ModalView
+    -- The source of truth for the unsaved change lives inside each editor
+    -- The only reason we store a copy of the state here is because of the render function
+    -- in the Mainframe view, which uses this derived state to show an unsaved indicator.
+    -- Inside the Mainframe view we can inspect the state of the submodules (Haskell/JS/Marlowe)
+    -- as their state is part of the MainFrame state, but we cannot inspect the state of the
+    -- child components Blockly and ActusBlockly as we need a Query.
```

Comment:
I think this comment needs updating, doesn't it? If I'm following correctly, this is now the _single_ source of truth, and is no longer copied in the editor states.